packages/calcite-components/src/tests/commonTests.ts

```diff
@@ -1376,3 +1380,204 @@ export async function t9n(componentTestSetup: ComponentTestSetup): Promise<void>
     await page.waitForChanges();
   }
 }
+
+interface BeforeToggle {
+  /**
+   * Function argument to simulate user input (mouse or keyboard), to open the component.
+   */
+  open: (page: E2EPage) => Promise<void>;
+
+  /**
+   * Function argument to simulate user input (mouse or keyboard), to close the component.
+   */
+  close: (page: E2EPage) => Promise<void>;
+}
+
+interface OpenCloseOptions {
+  /**
+   * Toggle property to test. Currently, either "open" or "expanded".
+   */
+  openPropName?: string;
+
+  /**
+   * Indicates the initial value of the toggle property to determine whether to configure a `simplePageSetup` or `newProgrammaticE2EPage`.
+   */
+  initialToggleValue?: boolean;
```

Comment:
Synced up w/ @Elijbet and we decided to see if we could eliminate the need for this prop by looking up the open prop name dynamically from the test HTML or component (separate PR).